Sources/Remote Logging/Crash Logging/PerformanceTracking.swift

```diff
@@ -0,0 +1,42 @@
+/// Defines whether to enable performance tracking, and if so, how to configure it.
+public enum PerformanceTracking {
+
+    case disabled
+    case enabled(Configuration)
```

Comment:
If you've seen the code in `CrashLogging` and `CrashLoggingDataProvider` above, you'll have noticed that the way one can configure performance monitoring in Sentry allows for inconsistent state, or at least _surprising_ state.

`enableAutoPerformanceTracking` is the main switch, if that's `false`, all the other settings are ignored. `tracesSampleRate` also needs to be more than `0.0` or no metric will be captured (fun fact: it defaults to 0.0 in the SDK, despite `enableAutoPerformanceTracking` defaulting to `true`). 

I suppose the Sentry authors decided to adopt this flat structure to preserve Objective-C compatibility. But we don't have to compromise like they do.

This approach, albeit quite verbose, makes undesirable state unrepresentable: For our clients, either performance tracking is `.enabled`, in which case they have a `Configuration` value, or `.disabled` in which case they have no configuration. It's then up to the library's internals to map down to what Sentry expects.